### DISCUSSION TITLE
ppstructure和paddleocr文本检测的效果差距较大

### DISCUSSION THREAD

**Discussion by bla-l** [PaddlePaddle/PaddleOCR#13659] [Q&A]
![7beec120413bbefd6ff4cc8a579d119](https://github.com/user-attachments/assets/ecef9935-6aba-45bb-bf88-79cf0a2897af)
![1dc6480dbe50933db2abd7d6e77b2df](https://github.com/user-attachments/assets/7e96d88d-0a27-43a7-871c-76ca6fe52612)
![34d01b5beb38d429ac596801b38928c](https://github.com/user-attachments/assets/f444aef3-8511-4a05-8201-0d3c8fc045f5)
第一张图是ppstructure的结果，第二张图是paddleocr的结果，第三张图是原图

**Comment by GreatV** [comment]
用的是啥版本呀，环境是为什么样的

**Reply by bla-l** [reply]
直接clone的main分支，Linux下跑的，模型啥的用的也是最新的版本

**Reply by GreatV** [reply]
![result](https://github.com/user-attachments/assets/cd95116a-4274-49bb-8c6c-1fdbbb00e6da)

这个是因为前面几行没有检测到layout block，被忽略了。

```text
text
title
figure
figure_caption
table
table_caption
header
footer
reference
equation

```

**Reply by bla-l** [reply]
那要如何修改呢😃

**Reply by GreatV** [reply]
可以试试把相应的参数阈值设小一点 `layout_score_threshold=0.1`

**Reply by bla-l** [reply]
好的我明天试试效果🌹，感谢

**Reply by bla-l** [reply]
设置完layout_score_threshold=0.1之后，他把之前没识别到的部分识别成table了，ocr依旧没有结果

**Reply by GreatV** [reply]
在table里面应该有内容

**Reply by bla-l** [reply]
那有什么办法让table的res和其他res一样保存在txt里吗，不想单独处理一次excel，而且table识别出来的结果也不对，少了一些文字。在其他的图片里也会遇到这种单独ocr结果很好，但是上了layout之后效果就会变差的情况
![80bda132f01e0bc2997a85fbb6a9a110](https://github.com/user-attachments/assets/7050200b-4c90-4938-860d-4606be2fe9d2)
![c8c04f02ffe7178635b5f03d7d041cba](https://github.com/user-attachments/assets/0300dcbf-b9a7-46ff-a673-9aa4cabd8cb6)
![7f8e1aaa525e5bbeaffaadabba80b094](https://github.com/user-attachments/assets/346d0130-1ad8-4d98-9a7f-8fd10f598295)
![c67ccef910cf6676835a66cc9a2661ac](https://github.com/user-attachments/assets/bf11aa98-3bc5-4b34-9621-441891c1ca3b)

**Reply by GreatV** [reply]
可以先ocr一遍，再检测layout，然后根据bbox信息把两者匹配起来

**Reply by bla-l** [reply]
这个我也想过，但有个问题就是如果layout没检测出来就没有layout的bbox，那这样的话应该如何与ocr的bbox匹配起来呢

**Reply by GreatV** [reply]
没检测到就没办法了，需要在自己的场景下重新训练一下模型。或者换其他检测模型试试。

**Reply by bla-l** [reply]
“换其他检测模型试试”是指换layout的模型吗，还是指det模型。det模型会影响layout的检测效果吗

**Reply by GreatV** [reply]
> “换其他检测模型试试”是指换layout的模型吗，还是指det模型。det模型会影响layout的检测效果吗

layout
